docs/DeveloperGuide.md

```diff
@@ -1,33 +1,212 @@
+* initialises two arraylists, `ArrayList<Grade>` and `ArrayList<People>` 
+to store the relevant `Grade` and `People` objects.
+* uses `AcademicCommandParser` to parse the user command.
+* identifies `AcademicCommandType` to decide 
+which of the commands under `PersonBook` or `GradeBook` is to be executed.
+* calls `AcademicStorage` to store the current set of data into the local storage file.
+
+**Academic Implementation**
+
+This section explains the details on how certain features 
+are implemented in the academic tracker.
+
+**GradeBook Features**
+The grade features are facilitated `Gradebook`, which further make use of `Grade`. 
+All grades are stored internally under `AcademicRun` as an array list `ArrayList<Grade>`.
+It implements the following operations:
+* `addGrade(String[], ArrayList<Grade>)`:Adds a `Grade` to the `ArrayList<Grade>`. 
+* `printCap(ArrayList<Grade>)`:Calculate the current CAP based on the `ArrayList<Grade>`. 
+* `printListOfGrades(ArrayList<Grade>)`:Print out all the `Grade` that are currently stored inside `ArrayList<Grade>`.
+* `deleteGrade(Integer, ArrayList<Grade>)`:Delete a `Grade` from a specified index inside `ArrayList<Grade>`.
+* `suGradeInGradeBook(Integer, ArrayList<Grade>)`:Su a `Grade` from a specified index inside `ArrayList<Grade>`.
+* `starGrade(Integer, ArrayList<Grade>)`:Star a `Grade` from a specified index inside `ArrayList<Grade>`.
+* `combineGradeDetails(Grade)`:Print out the details of a particular `Grade`.
+
+### **Flashcard Component**
+
+This section will describe in detail how the flashcard feature is implemented.
+
+![Flashcard Component Class Diagram](Images/Flashcard_Class.png)
+
+The above diagram looks at the overall structure of how the flashcard component is being implemented. 
+This component is split into 4 different classes, their associations and multiplicity as explained in 
+the above Figure 7. As shown in Figure 7, the main class `FlashcardRun` will be accessed when the 
+flashcard mode is called in `StudyIt` Class. `FlashcardRun` class is associated with `FlashcardStorage` 
+class that is used to store data in .txt file and `FlashcardDeck` class which contains any number of `Flashcards`.
+
+**Flashcard Implementation**
+
+![Sequence Diagram when user input “add card” command](Images/addCard_sequenceDiagram.png)
+
+With reference to Figure 8, when `FlashcardRun` is first initialised by `StudyIt`, it will construct 
+the `FlashcardDeck` class. As an add card command is given by the user, `FlashcardRun` will take in the 
+command and call `addCard()` method in `FlashcardDeck` which constructs a new Flashcard object and stores 
+it inside the `FlashcardDeck` object.The `addCard()` function will then show the user the question and 
+answer of the flashcard that have been created.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+##Documentation, logging, testing, configuration, dev-ops
+###Testing guide
```

Comment:
The headers are not rendered correctly in Github Pages. You might want to add a space.
```
### Documentation
instead of
###Documentation
```